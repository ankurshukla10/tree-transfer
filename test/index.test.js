import React from 'react';
import { render, mount } from 'enzyme';
import TreeTransfer from '../src';

const treeTransferProps = {
  source: [
    {
      key: '0',
      title: '0',
      children: [
        {
          key: '0-0',
          title: '0-0',
        },
        {
          key: '0-1',
          title: '0-1',
        }
      ]
    }
  ],
  target: ['0-1']
};

describe('TreeTransfer', () => {
  it('renders correctly', () => {
    const wrapper = render(<TreeTransfer {...treeTransferProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should move all selected tree leafs to right list', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<TreeTransfer {...treeTransferProps} onChange={handleChange} />);
    wrapper.find(TreeTransfer).find('.ant-tree').find('.ant-tree-checkbox').at(0).simulate('click');
    wrapper.find(TreeTransfer).find('.tree-transfer-operation').find('button').at(0).simulate('click');
    expect(handleChange).toHaveBeenLastCalledWith(['0-0', '0-1']);
  });

  it('should move unselected tree leafs to right list', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<TreeTransfer {...treeTransferProps} onChange={handleChange} />);
    wrapper.find(TreeTransfer).find('.ant-tree').find('.ant-tree-checkbox').at(2).simulate('click');
    wrapper.find(TreeTransfer).find('.tree-transfer-operation').find('button').at(0).simulate('click');
    expect(handleChange).toHaveBeenLastCalledWith([]);
  });
});
  