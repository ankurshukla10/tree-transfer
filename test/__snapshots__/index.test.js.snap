// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeTransfer renders correctly 1`] = `
<div
  class="lucio-tree-transfer"
>
  <div
    class="tree-transfer-panel tree-transfer-panel-left"
  >
    <div
      class="tree-transfer-panel-header"
    >
      <span
        class="tree-transfer-panel-header-select"
      >
        1/2 条数据
      </span>
      <span
        class="tree-transfer-panel-header-title"
      >
        源数据
      </span>
    </div>
    <div
      class="tree-transfer-panel-body"
    >
      <div
        class="lucio-loading"
      >
        <div
          class="lucio-loading-content"
        >
          <div
            class="tree-transfer-panel-body-content"
          >
            <ul
              class="ant-tree"
              role="tree-node"
              unselectable="on"
            >
              <li
                class=""
              >
                <span
                  class="ant-tree-switcher ant-tree-switcher_open"
                />
                <span
                  class="ant-tree-checkbox ant-tree-checkbox-indeterminate"
                >
                  <span
                    class="ant-tree-checkbox-inner"
                  />
                </span>
                <span
                  class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-open"
                  title="0"
                >
                  <span
                    class="ant-tree-title"
                  >
                    0
                  </span>
                </span>
                <ul
                  class="ant-tree-child-tree ant-tree-child-tree-open"
                  data-expanded="true"
                >
                  <li
                    class=""
                  >
                    <span
                      class="ant-tree-switcher ant-tree-switcher-noop"
                    />
                    <span
                      class="ant-tree-checkbox"
                    >
                      <span
                        class="ant-tree-checkbox-inner"
                      />
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
                      title="0-0"
                    >
                      <span
                        class="ant-tree-title"
                      >
                        0-0
                      </span>
                    </span>
                  </li>
                  <li
                    class=""
                  >
                    <span
                      class="ant-tree-switcher ant-tree-switcher-noop"
                    />
                    <span
                      class="ant-tree-checkbox ant-tree-checkbox-checked"
                    >
                      <span
                        class="ant-tree-checkbox-inner"
                      />
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
                      title="0-1"
                    >
                      <span
                        class="ant-tree-title"
                      >
                        0-1
                      </span>
                    </span>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="tree-transfer-operation"
  >
    <button
      class="ant-btn ant-btn-primary ant-btn-sm ant-btn-icon-only"
      disabled=""
      type="button"
    >
      <i
        class="anticon anticon-right"
      />
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-sm ant-btn-icon-only"
      disabled=""
      type="button"
    >
      <i
        class="anticon anticon-left"
      />
    </button>
  </div>
  <div
    class="tree-transfer-panel tree-transfer-panel-right"
  >
    <div
      class="tree-transfer-panel-header"
    >
      <label
        class="ant-checkbox-wrapper"
      >
        <span
          class="ant-checkbox"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
      <span
        class="tree-transfer-panel-header-select"
      >
        1 条数据
      </span>
      <span
        class="tree-transfer-panel-header-title"
      >
        目的数据
      </span>
    </div>
    <div
      class="tree-transfer-panel-body"
    >
      <ul
        class="tree-transfer-panel-body-content"
      >
        <li>
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span>
            0-1
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
